source/pkcs11.c

```diff
@@ -398,3 +426,312 @@ void aws_pkcs11_lib_release(struct aws_pkcs11_lib *pkcs11_lib) {
         aws_ref_count_release(&pkcs11_lib->ref_count);
     }
 }
+
+/**
+ * Find the slot that meets all criteria:
+ * - has a token
+ * - if match_slot_id is non-null, then slot IDs must match
+ * - if match_token_label is non-null, then labels must match
+ * The function fails unless it finds exactly one slot meeting all criteria.
+ */
+int aws_pkcs11_lib_find_slot_with_token(
+    struct aws_pkcs11_lib *pkcs11_lib,
+    const aws_pkcs11_t *match_slot_id,
+    const struct aws_string *match_token_label,
+    aws_pkcs11_t *out_slot_id) {
+
+    CK_SLOT_ID *slot_id_array = NULL;
+    CK_SLOT_ID *candidate = NULL;
+    CK_TOKEN_INFO info;
```

Comment:
Dont you need to zero initialize this?

Reply:
Good idea
I was assuming the PKCS#11 lib would do that, but better to be safe